main.R

```diff
@@ -136,6 +136,42 @@ if (nrow(techexposure_data) > 0) {
 techexposure_data
   }
 
+prep_exposure_stats <- function(audit_file, investor_name, portfolio_name, pacta_sectors) {
+  audit_table <- pacta.portfolio.report:::prep_audit_table(
+      audit_file,
+      investor_name = investor_name,
+      portfolio_name = portfolio_name,
+      currency_exchange_value = currency_exchange_value
+    )
+  
+  exposure_stats <- audit_file %>%
+      filter(.data$investor_name == .env$investor_name &
+        .data$portfolio_name == .env$portfolio_name) %>%
+      filter(.data$asset_type %in% c("Bonds", "Equity")) %>%
+      filter(.data$valid_input == TRUE) %>%
+      mutate(across(c("bics_sector", "financial_sector"), as.character)) %>%
+      mutate(
+        sector =
+          if_else(!.data$financial_sector %in% .env$pacta_sectors,
+            "Other",
+            .data$financial_sector
+          )
+      ) %>%
+      group_by(.data$asset_type, .data$sector) %>%
+      summarise(
+        value = sum(.data$value_usd, na.rm = TRUE) / .env$currency_exchange_value,
+        .groups = "drop"
+      ) %>%
```

Comment:
NB: I'm pretty you can use the `summarise(..., .by = c("asset_type", "sector"))` syntax here (same with the `mutate()` call later. It is a slightly more concise way to do the `group_by`, `do_something`, `ungroup` pattern.

Reply:
fair, I didn't know that one